daemon/src/db.rs

```diff
@@ -367,6 +623,409 @@ pub struct Cfd {
     pub initial_tx_fee_rate: TxFeeRate,
 }
 
+/// Data loaded from the database about a closed CFD.
+#[derive(Debug, Clone, Copy)]
+pub struct ClosedCfd {
+    pub id: OrderId,
+    pub position: Position,
+    pub initial_price: Price,
+    pub taker_leverage: Leverage,
+    pub n_contracts: Contracts,
+    pub counterparty_network_identity: Identity,
+    pub role: Role,
+    pub fees: Fees,
+    pub expiry_timestamp: OffsetDateTime,
+    pub lock: Lock,
+    pub settlement: Settlement,
+}
+
+/// Data loaded from the database about the lock transaction of a
+/// closed CFD.
+#[derive(Debug, Clone, Copy)]
+pub struct Lock {
+    pub txid: Txid,
+    pub dlc_vout: Vout,
+}
+
+/// Data loaded from the database about the way in which a closed CFD
+/// was settled.
+///
+/// It is represented using an `enum` rather than a series of optional
+/// fields so that only sane combinations of transactions can be
+/// loaded from the database.
+#[derive(Debug, Clone, Copy)]
+pub enum Settlement {
+    Collaborative {
+        txid: Txid,
+        vout: Vout,
+        payout: Payout,
+        price: Price,
+    },
+    Cet {
+        commit_txid: Txid,
+        txid: Txid,
+        vout: Vout,
+        payout: Payout,
+        price: Price,
+    },
+    Refund {
+        commit_txid: Txid,
+        txid: Txid,
+        vout: Vout,
+        payout: Payout,
+    },
+}
+
+/// All the data related to a closed CFD that we want to store in the
+/// database.
+#[derive(Debug, Clone, Copy)]
+struct ClosedCfdInput {
+    id: OrderId,
+    position: Position,
+    initial_price: Price,
+    taker_leverage: Leverage,
+    n_contracts: Contracts,
+    counterparty_network_identity: Identity,
+    role: Role,
+    fees: Fees,
+    expiry_timestamp: OffsetDateTime,
+    lock: LockInput,
+    collaborative_settlement: Option<CollaborativeSettlement>,
+    commit: Option<Txid>,
+    non_collaborative_settlement: Option<Cet>,
+    refund: Option<Refund>,
+}
+
+/// Auxiliary type used to gradually combine a `Cfd` with its list of
+/// `CfdEvent`s.
+///
+/// Once all the `CfdEvent`s have been applied, we can build a
+/// `ClosedCfdInput` which is used for database insertion.
+#[derive(Debug, Clone)]
+struct ClosedCfdInputAux {
```

Comment:
For me this is the same concept as what we called `...Aggregate` in e.g. the projection.
If you agree we might want to align the name. Not blocking though :)

Reply:
It is an internal aggregate of the db, true~ I can call it `ClosedCfdInputAggregate`.